server/planning/feeding_services/onclusive_api_service.py

```diff
@@ -0,0 +1,117 @@
+import logging
+from datetime import datetime, date, timedelta
+import requests
+from superdesk.io.feeding_services.http_base_service import HTTPFeedingServiceBase
+
+logger = logging.getLogger(__name__)
+
+
+class OnclusiveApiService(HTTPFeedingServiceBase):
+    """
+    Feeding Service class which can read events using HTTP
+    """
+
+    NAME = "Onclusive_api"
+    label = "Onclusive api feed"
+    service = "events"
+    fields = [
+        {
+            "id": "url",
+            "type": "text",
+            "label": "Feed URL",
+            "placeholder": "Feed URL",
+            "required": True,
+            "default": "https://api.forwardplanner.com/",
+        },
+        {"id": "username", "type": "text", "label": "Username", "placeholder": "Username", "required": True},
+        {"id": "password", "type": "password", "label": "Password", "placeholder": "Password", "required": True},
+    ]
+
+    HTTP_AUTH = False
+
+    def _update(self, provider, update):
+        """
+        Fetch events from external API.
+
+        :param provider: Ingest Provider Details.
+        :type provider: dict
+        :param update: Any update that is required on provider.
+        :type update: dict
+        :return: a list of events which can be saved.
+        """
+        session = requests.Session()
+        current_date = datetime.now()
+        URL = provider["config"]["url"]
+        USERNAME = provider["config"]["username"]
+        PASSWORD = provider["config"]["password"]
+        TOKEN = ""
+        REF_TOKEN = ""
+        # authntication get token
+        if not TOKEN:
```

Comment:
this will be always true we don't keep instances around but we always create a new one

Reply:
but it might make sense to store it in db (probably not the token but the refresh one which lasts for a few days)